fnl/hotpot/searcher/module.fnl

```diff
@@ -154,14 +154,12 @@
   ;; If stale or missing, complile and return a loader for the cached file
   ;; If the original modname was for a lua file, just return a loader for that.
 
-  ;; TODO unsure about this double nested form but if feels problematic
-  (match (. package :preload modname)
-    module module
-    nil (match (modname-to-path modname)
-          ;; modpath can be nil if no file is found for modname
-          nil nil
-          path (match (pcall create-loader! modname path)
-                 (true loader) loader
-                 (false errors) (create-error-loader modname path errors)))))
+  (or (. package :preload modname)
+      (match (modname-to-path modname)
+        ;; modpath can be nil if no file is found for modname
+        nil nil
```

Comment:
`nil nil` actually implied since `path` cant be nil but maybe better to be explicit for reading